x-pack/plugins/apm/public/components/app/profiling_overview/profiling_flamegraph.tsx

```diff
@@ -86,11 +97,24 @@ export function ProfilingFlamegraph({
         </EuiFlexItem>
       </EuiFlexGroup>
       <EuiSpacer />
-      <EmbeddableFlamegraph
-        data={data?.flamegraph}
-        isLoading={isPending(status)}
-        height="60vh"
-      />
+      {status === FETCH_STATUS.SUCCESS && isEmpty(data) ? (
+        <EuiEmptyPrompt
+          titleSize="s"
+          title={
+            <div>
+              {i18n.translate('xpack.apm.profiling.flamegraph.noDataFound', {
+                defaultMessage: 'No data found',
```

Comment:
is this something usually expected? or a way to not crash whenever the user enters the app route directly?
I'm asking because it would be interesting, if this is something usually expected, to have something pointing out the user on how to get UP data, and if it's just for protecting the route maybe pointing to the documentation where we have the unsupported types

Reply:
To filter profiling data I need the service host.name field. So in case there's no host.name available, like for RUM agents there'll be no data to be shown here.

Reply:
IIUC users would likely not get into this state unless they use directly an url 👍🏼